crates/integration_test/src/utils/test_utils.rs

```diff
@@ -64,62 +64,65 @@ pub fn create_container<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<Child> {
 }
 
 /// Sends a kill command to the given container process
-pub fn kill_container<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<Child> {
-    let res = Command::new(get_runtime_path())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .arg("--root")
-        .arg(dir.as_ref().join("runtime"))
+pub fn kill_container<P: AsRef<Path>>(id: &str, dir: P) -> Result<Child> {
+    let res = runtime_command(dir)
         .arg("kill")
-        .arg(id.to_string())
+        .arg(id)
         .arg("9")
         .spawn()
         .context("could not kill container")?;
     Ok(res)
 }
 
-pub fn delete_container<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<Child> {
-    let res = Command::new(get_runtime_path())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .arg("--root")
-        .arg(dir.as_ref().join("runtime"))
+pub fn delete_container<P: AsRef<Path>>(id: &str, dir: P) -> Result<Child> {
+    let res = runtime_command(dir)
         .arg("delete")
-        .arg(id.to_string())
+        .arg(id)
         .spawn()
         .context("could not delete container")?;
     Ok(res)
 }
 
-pub fn get_state<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<(String, String)> {
-    sleep(SLEEP_TIME);
-    let output = Command::new(get_runtime_path())
+pub fn get_state<P: AsRef<Path>>(id: &str, dir: P) -> Result<Child> {
+    let res = Command::new(get_runtime_path())
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .arg("--root")
         .arg(dir.as_ref().join("runtime"))
         .arg("state")
         .arg(id.to_string())
-        .spawn()?
-        .wait_with_output()?;
+        .spawn()
+        .context("could not get container state")?;
+    Ok(res)
+}
+
+pub fn get_state_output<P: AsRef<Path>>(id: &str, dir: P) -> Result<(String, String)> {
```

Comment:
I don't have any better suggestions right now either, but can we change the names of these functions? `get_state` name indicates that it will return the state of the container rather than the child process running `runtime state` command.

One suggestion is as far as I can see, current `get_state` is only used in the state test, and as per my comment on it, if we can replace it with current `get_state_output` then we can make the current `get_state` a private function of this module, and rename to something like `run_state_command` and rename `get_state_output` to `get_state`

Reply:
I have ended up removing the `get_state` function completely and just renamed `get_state_output`->`get_state` as after refactoring `lifecycle/state.rs` it was no longer needed.